hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssShareSignatureHandler.java

```diff
@@ -90,8 +89,7 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         final var tssSignaturesMap = signatures.computeIfAbsent(messageHash, k -> new ConcurrentHashMap<>());
         final Set<TssShareSignature> tssShareSignatures =
                 tssSignaturesMap.computeIfAbsent(rosterHash, k -> ConcurrentHashMap.newKeySet());
-        final var isPresent =
-                tssShareSignatures.stream().anyMatch(sig -> sig.shareId().idElement() == shareIndex);
+        final var isPresent = tssShareSignatures.stream().anyMatch(sig -> sig.shareId() == shareIndex);
```

Comment:
Where is `shareIndex` coming from? 
We need to consider that shareIds and the positional location of the elements on a sorted list by share id is off by 1.

Reply:
It is set in the transaction body from [here](https://github.com/hashgraph/hedera-services/blob/16587-library-integration-v1/hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssBaseServiceImpl.java#L241). It is taking shareId from `TssPrivateShare`. I am assuming that will be correct shareId.